gcp/modules/gke-cluster/main.tf

```diff
@@ -28,24 +28,13 @@ module "gke_cluster" {
   project_id         = "${var.project_id}"
   serviceaccount_key = "${var.serviceaccount_key}"
 
-  initial_node_count = "${var.initial_node_count}"
-  node_type          = "${var.node_type}"
-
   kubernetes_version = "1.11.6-gke.3"
 
   region = "${var.infra_region}"
 
   monitoring_service = "monitoring.googleapis.com/kubernetes"
   logging_service    = "logging.googleapis.com/kubernetes"
 
-  oauth_scopes = [
-    "https://www.googleapis.com/auth/compute",
-    "https://www.googleapis.com/auth/devstorage.read_only",
-    "https://www.googleapis.com/auth/logging.write",
-    "https://www.googleapis.com/auth/monitoring",
-    "https://www.googleapis.com/auth/trace.append",
```

Comment:
If I'm understanding correctly, you're replacing several specific scopes (here) with one very broad scope (`cloud-platform`)? Is this desirable?

Reply:
Yes, and in the end, this will result in more narrow permissions for individual pods.

Atm. any pod on the node can access node service account with `editor` role on the whole project and the above-mentioned scopes (plus `compute-rw` one, another exception - this is how k8s snapshots are able to work for example).

After deploying the service-account-assigner, only workloads running directly on the node and service-assigner pods will be able to use this account, all other pods will be restricted to separate "pod default" account with much more limited permissions or use their individual specific accounts, this will be done via annotations like:
```
  accounts.google.com/service-account: "${service_account}"
  accounts.google.com/scopes: "ndev.clouddns.readonly,ndev.clouddns.readwrite"
```

2nd part of the answer is that scopes are a legacy access control method (see https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances#using_the_compute_engine_default_service_account best practices section, or https://cloud.google.com/compute/docs/access/service-accounts#accesscopesiam). The recommended way is to set `cloud-platform` scope, and properly restrict permissions via IAM for the used service account (and we're moving away from acount with `editor` role here).

Reply:
Great, thanks for the explanation and the links.